tracer/src/Datadog.Trace/AppSec/Waf/Waf.cs

```diff
@@ -66,53 +78,107 @@ public void Dispose(bool disposing)
         {
+            if (!File.Exists(rulesFile))
+            {
+                Log.Error($"AppSec could not find the rules file in path \"{rulesFile}\". AppSec will not run any protections in this application.");
+                return null;
+            }
+
+            return File.OpenRead(rulesFile);
+        }
+
+        // null rulesFile means use rules embedded in the manifest
+        private static WafHandle CreateWafHandle(string rulesFile)
+        {
+            var argCache = new List<Obj>();
+            Obj configObj;
             try
             {
-                DdwafConfigStruct args = default;
+                using var stream = GetRulesStream(rulesFile);
 
-                var argCache = new List<Obj>();
-                var rules = RuleSetFromManifest(argCache);
+                if (stream == null)
+                {
+                    return null;
+                }
 
-                var ruleHandle = WafNative.Init(rules.RawPtr, ref args);
+                configObj = CreatObjFromRulesStream(argCache, stream);
+            }
+            catch (Exception ex)
+            {
+                var message =
+                    rulesFile != null ?
+                    $"AppSec could not read the rule file \"{rulesFile}\" as it was invalid. AppSec will not run any protections in this application." :
+                    "AppSec could not read the rule file emmbeded in the manifest as it was invalid. AppSec will not run any protections in this application.";
+                Log.Error(ex, message);
 
-                rules.Dispose();
+                return null;
+            }
+
+            try
+            {
+                DdwafConfigStruct args = default;
+                var ruleHandle = WafNative.Init(configObj.RawPtr, ref args);
+                return new WafHandle(ruleHandle);
+            }
+            finally
+            {
+                configObj?.Dispose();
                 foreach (var arg in argCache)
                 {
                     arg.Dispose();
                 }
+            }
+        }
 
-                if (ruleHandle == IntPtr.Zero)
-                {
-                    Log.Error("Failed to create rules.");
-                    return null;
-                }
-                else
+        private static void LogRuleDetails(JToken root)
+        {
+            try
+            {
+                var eventsProp = root.Value<JArray>("events");
+                foreach (var ev in eventsProp)
                 {
-                    Log.Information("Rules successfully created.");
+                    var idProp = ev.Value<JValue>("id");
+                    var nameProp = ev.Value<JValue>("name");
+                    var addresses = ev.Value<JArray>("conditions").SelectMany(x => x.Value<JObject>("parameters").Value<JArray>("inputs"));
+                    Log.Debug($"Loaded rule: {idProp.Value} - {nameProp.Value} on addresses: {string.Join(", ", addresses)}");
```

Comment:
For this one, we can check if debug logging is enabled before concatenating addresses.

```suggestion
                    if(Log.IsEnabled(Datadog.Trace.Vendors.Serilog.Events.LogEventLevel.Debug))
                    {
                        Log.Debug("Loaded rule: {id} - {name} on addresses: {addresses}", idProp.Value, nameProp.Value, string.Join(", ", addresses));
                    }
```

Reply:
this whole method is only called if debug is enabled, I've refactored in 1c0de753d to make that clearer.

Reply:
Oh, nice. Sorry I didn't catch that.